### PR TITLE
(WIP) update api doc

### PR DESCRIPTION
**Pull request by zahodi**

### CODE REVIEW

File: docs/api.adoc [hunk 1]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
...
-  /api/host/list
+GET /api/search?type=any&name=web_hosts
 ....
 ** Example results:
 +
 [source,json]
 ----
 {
+  "status": "success",
+  "code": 200,
```
Comment (senorsmile) [inline docs/api.adoc]: http status codes should not be in the JSON too, since they'll already be in the HTTP header

File: docs/api.adoc [hunk 2]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
...
-
 [nested]
-* List all
+* Search for host or group.
+Specify type as  a group, host, or any to limit the search.
 +
 [source]
 ....
-  /api/host/list
+GET /api/search?type=any&name=web_hosts
```
Comment (senorsmile) [inline docs/api.adoc]: based on a search term of "web_hosts", the below groups and hosts don't actually make any sense.  Let's modify the example response to show that this is a regex that will match.

File: docs/api.adoc [hunk 3]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
...
+
+* Delete a host
+[source]
+....
+DELETE /api/hostvars?name=name_of_the_host
+....
 
 Groups
 -----
+** Example response:
```
Comment (senorsmile) [inline docs/api.adoc]: are we missing the example query here?

File: docs/api.adoc [hunk 4]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
...
++
+[source,json]
+----
+{
+  "status": "successful",
+  "code": 200,
+  "data": {
+    "group_name": "group1",
+    "parent_group": "all",
```
Comment (senorsmile) [inline docs/api.adoc]: there will never be a parent_group called all.  for so called "top_level_groups", there will be a key called "parent_groups": []  assigned to an empty list.  

We should also show "children": [ ] here with a list of children groups.

File: docs/api.adoc [hunk 5]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
...
 ----
 {
-  "hosts": [
-    "host1",
-    "host10"
-  ]
+  "status": "successful",
+  "code": 200,
+  "data": {
+    "parent_group": "group1",
```
Comment (senorsmile) [inline docs/api.adoc]: this should be "parent_groups"

and it should have a list, not a scalar.

File: docs/api.adoc [hunk 6]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
...
+    "host": "host1",
```
Comment (senorsmile) [inline docs/api.adoc]: this should be called "hosts",

and it should have a list, not a scalar

File: docs/api.adoc [hunk 7]
```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
 
 Detailed description of api docs here!
 
-Hosts
+Search
```
Comment (senorsmile) [inline docs/api.adoc]: Let's call out the headers like this:

GET Search
GET Hostvars etc. to be consistent and clear in the menu at the top.

File: docs/api.adoc (hunk 6 above)
Comment (zahodi) [reply docs/api.adoc]: a host can not contain hosts. In this response "host" is the name of the the host that we are querying for.

File: docs/api.adoc [hunk 8]
```diff
@@ -7,58 +7,197 @@ v2.0.0, 2016-12
...
+    "children": [
+      "group6",
+      "group20"
+    ],
+  }
+}
+----
+** HTTP response code: 200
+
+Get top level groups
```
Comment (senorsmile) [inline docs/api.adoc]: Capitalize Get to GET for consistency

File: docs/api.adoc [hunk 9]
```diff
@@ -7,69 +7,314 @@ v2.0.0, 2016-12
...
-    "host1",
-    "host2",
-    "host9",
-    "host10"
+   "webhost1"
+   "webhost2"
   ]
 }
 ----
+** HTTP response code: 200
```
Comment (senorsmile) [inline docs/api.adoc]: Maybe change these lines to: 

```Normal HTTP respond code: 200```

File: docs/api.adoc [hunk 10]
```diff
@@ -7,69 +7,314 @@ v2.0.0, 2016-12
...
-* Currently, only v2.0 of this api is available.
-As new versions come out, we will strive for backward compatibility, and
-will always "attempt" to support previous versions of the api.
-* List all with api version
+
+
+
+POST /api/hostvars
+~~~~~~~~~~~~~~~~~~
+[source]
```
Comment (senorsmile) [inline docs/api.adoc]: add a description to POST /api/hostvars

**Comment by senorsmile** [comment]
👍

**Pull request merged**